adapters/http-base/src/main/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapter.java

```diff
@@ -632,27 +628,25 @@ private void doUploadMessage(
                             .onSuccess(effectiveTtd -> Optional.ofNullable(effectiveTtd)
                                     .ifPresent(v -> TracingHelper.TAG_DEVICE_TTD.set(currentSpan, v)));
                 });
-        final Future<CommandConsumer> commandConsumerTracker = ttdTracker
-                .compose(ttd -> createCommandConsumer(
+        final Future<ResponseReadyTracker> responseReadyTrackerFuture = ttdTracker
+                .compose(ttd -> createCommandTracker(
                         ttd,
                         tenantTracker.result(),
                         deviceId,
                         gatewayId,
                         ctx.getRoutingContext(),
-                        responseReady,
                         currentSpan));
 
-        commandConsumerTracker
-        .compose(ok -> {
+        responseReadyTrackerFuture
+        .compose(responseReadyTracker -> {
 
             final Map<String, Object> props = getDownstreamMessageProperties(ctx);
-            Optional.ofNullable(commandConsumerTracker.result())
+            Optional.ofNullable(responseReadyTrackerFuture.result())
```

Comment:
`responseReadyTrackerFuture` is only succeeded with a non-null value, so `responseReadyTrackerFuture.result()` (i.e. `responseReadyTracker`) is never null here.